pkg/controllers/managementuser/nodesyncer/nodessyncer.go

```diff
@@ -135,9 +137,69 @@ func (n *nodeSyncer) sync(key string, node *corev1.Node) (runtime.Object, error)
 		n.machines.Controller().Enqueue(n.clusterNamespace, AllNodeKey)
 	}
 
+	cluster, err := n.nodesSyncer.clusterLister.Get("", n.clusterNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	var (
+		updateVersion string
+	)
+
+	// only applies to imported k3s/rke2 clusters
+	if cluster.Status.Driver == apimgmtv3.ClusterDriverK3s {
+		if cluster.Spec.K3sConfig == nil {
+			return nil, nil
```

Comment:
```suggestion
			return cluster, nil
```

Reply:
We definitely don't want to return `cluster, nil` in the node syncer.

Reply:
Could you explain a little why?  How is the sync handler different from the general ObjectHandler?

Reply:
This is a handler for downstream nodes, not cluster objects, just check the function signature.